src/main/java/ladder/domain/ladder/Line.java

```diff
@@ -3,33 +3,37 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
-import ladder.domain.ladder.generator.BooleanGenerator;
+import ladder.domain.ladder.generator.RungGenerator;
 
 public class Line {
-    private final BooleanGenerator booleanGenerator;
+    private final RungGenerator rungGenerator;
     private final List<Rung> rungs;
 
-    public Line(final int playerCount, final BooleanGenerator booleanGenerator) {
-        this.booleanGenerator = booleanGenerator;
+    public Line(int playerCount, RungGenerator rungGenerator) {
+        this.rungGenerator = rungGenerator;
         this.rungs = generateRungs(playerCount);
     }
 
-    public List<Rung> generateRungs(final int playerCount) {
-        final List<Rung> rungs = new ArrayList<>();
+    public List<Rung> generateRungs(int playerCount) {
+        List<Rung> rungs = new ArrayList<>();
 
         for (int i = 0; i < playerCount - 1; i++) {
-            final Rung rung = generateRung(rungs, i);
+            Rung rung = generateRung(rungs);
             rungs.add(rung);
         }
 
         return rungs;
     }
 
-    private Rung generateRung(List<Rung> rungs, int index) {
-        boolean previousRungExist = index > 0 && rungs.get(index - 1).isExist();
-        boolean canMakeRung = booleanGenerator.generate();
+    private Rung generateRung(List<Rung> rungs) {
+        boolean previousRungExist = !rungs.isEmpty() && rungs.get(rungs.size() - 1).isExist();
+        Rung currentRung = rungGenerator.generate();
 
-        return Rung.of(!previousRungExist && canMakeRung);
+        if (previousRungExist || !currentRung.isExist()) {
```

Comment:
부정연산자가 가독성을 떨어뜨린다는 의견이 있어서 currentRung.notExist() 를 만들어주는 방법도 있습니다

Reply:
그렇군요!
 메서드가 늘어나는 것이 안좋다고 생각했었는데 isEmpty()같은 것을 사용하면 가독성이 높아지겠군요!

Reply:
아하 죄송해요ㅜ suggestion으로 했어야 의도가 정확했겠네요
```suggestion
        if (previousRungExist || currentRung.notExist()) {
```
위 처럼 하는게 어떤가 하는 의견이었어요.
물론 검사식 자체를 메서드로 추출하는 것도 좋습니다
```suggestion
        if (isExist()) {
```

public 메서드가 늘어나는건 안 좋아요. 객체 역할이 많아지고 있다는 신호이고, 역할이 많다 == 여러사람이 동시에 수정할 일이 많아진다 == 컨플릭트가 늘어난다 입니다. 또 역할이 많으니 클라이언트도 많아지고, 나중엔 손도 못대는 코드가 됩니다 ㅎ ㅎ (모든 클라이언트의 정상 동작을 보장해줘야 하니까 겁나서 수정을 못 함)

근데 private 개수는 딱히 상관 없는거 같아요

Reply:
아 이해했어요. 제가 위에 적었던 isEmpty()가 Rung Enum에 있는 메서드를 말하는거였어요.
```java
package ladder.domain.ladder;

public enum Rung {
    EXIST,
    EMPTY;

    public static Rung of(boolean exist) {
        if (exist) {
            return EXIST;
        }
        return EMPTY;
    }

    public boolean isExist() {
        return this == EXIST;
    }

    public boolean isEmpty() {
        return this == EMPTY;
    }
}

```